corehq/apps/settings/forms.py

```diff
@@ -54,8 +56,19 @@ def __init__(self, user, *args, **kwargs):
             ),
         )
 
-    def clean_new_password1(self):
-        return clean_password(self.cleaned_data.get('new_password1'))
+    def clean_old_password(self):
+        from corehq.apps.hqwebapp.utils import decode_password
+        old_password = decode_password(self.cleaned_data['old_password'])
+        if old_password == '':
+            raise ValidationError(
+                _("Password cannot be empty"), code='password_empty',
+            )
+        if not self.user.check_password(old_password):
+            raise forms.ValidationError(
+                self.error_messages['password_incorrect'],
+                code='password_incorrect',
+            )
```

Comment:
> It's surprising to me that we need to raise these two validation errors in this PR since they should be handled at other places in the code.

`if not self.user.check_password(old_password)` is to check if user entered their current password correctly. This is a password reset form. This would not be used anywhere else.

I think we can effectively just keep the `check_password`. In case of an empty password which would happen only in case someone messes up the hashing and then would get the message "Incorrect password" instead of "empty password"